### DISCUSSION TITLE
Retrying RACH for Multiple RRC Connections

### DISCUSSION THREAD

**Discussion by Kong-TaeHyeon** [srsran/srsRAN_4G#1433] [General]
#### Goal
I want to establish multiple RRC connections with the same eNB.
I have set up the USRP B210 as the eNB and the USRP B205 mini as the UE.
I have modified the handle_con_setup function (src/stack/rrc/rrc.cc) and the parse_authentication_response function (src/stack/upper/nas.cc).

![image](https://github.com/user-attachments/assets/66435828-0293-41d0-af4d-dc460a74ea7b)
![image](https://github.com/user-attachments/assets/8f90d269-effa-43f6-b28f-6aa120fa80c6)

#### My Intention
After the UE successfully establishes an RRC connection with the eNB, handle_con_setup is invoked, but the NAS authentication response is not sent, causing the eNB to wait. The UE then calls leave_connection(). After this, the UE retries the RACH and RRC procedures. This process repeats.

#### Problem (I Guess. . .)
However, when the UE retries RACH, it seems to perform a cell search again, which causes a delay of about 20 seconds before the connection is re-established. How can I retry RACH without performing the cell search?
![image](https://github.com/user-attachments/assets/6622c57b-4cf7-4172-a60f-4e2d99c6c9b9)
![image](https://github.com/user-attachments/assets/b920df13-8731-4744-a5e8-000b6d71ab24)

**Comment by caseyonline** [comment]
Is a 'Connection Setup Complete' message being sent from the UE?   It's my understanding that this needs to occur before the eNB (HSS) sends an 'Authentication Request'.

**Reply by Kong-TaeHyeon** [reply]
I think that 'Connection Setup Complete' Message is included 'Authentication Request' (`handle_con_setup` -> `send_authentcation_response`).  In my situation (Just Multiple RRC Connection, Don't need NAS. . . .), UE don't send 'Connection Setup Complete' message. Is my method consistent with my goals?
